docs/fxa-engineering/fxa-webchannel-protocol.md

```diff
@@ -0,0 +1,324 @@
+
+For WebChannel usage in the Fenix browser, see [FxA/OAuth WebChannel Protocol](./fxa-oauth-webchannel-protocol).
+
+### Context
+
+WebChannels are used to communicate Sync related information when
+`service=sync&context=<context>` query parameters are specified.
+`<context>` should be one of:
+
+- `fx_desktop_v3`
+- `fx_fennec_v1`
+
+### Commands
+
+A command is a `WebChannelMessageToChrome` custom event sent from FxA to the browser using the `account_updates` WebChannel.
+
+Command format is explained in the section titled [Command/Request format](#command-request-format).
+
+#### fxaccounts:fxa_status
+
+Sent on startup to fetch the FxA state from supporting browsers.
+Used to get the browser's currently signed in user as well as browser capabilities.
+
+##### request data
+
+```js
+{
+  isPairing: <boolean>
+  service: <string>
+}
+```
+
+- `isPairing`: whether or not accounts.firefox.com is executing a pairing flow
+- `service`: the service name or OAuth client-id that is requesting authentication
+
+##### response data
+
+```js
+{
+  capabilities: <capabilities>
+  signedInUser: <signed in user> || null
+}
+```
+
+- `capabilities` - A list of capabilities supported by the browser. See [capabilities](#capabilities).
+- `signedInUser` - the user currently signed into Sync on the browser. See [signedInUser](#signedInUser).
+
+##### support
+
+Fx Desktop 55+
+
+#### fxaccounts:loaded
+
+Sent on startup. Sent after the first screen is visible. No data is sent. No response is expected.
+
+#### fxaccounts:can_link_account
+
+Sent on signin or signup after a user has entered their credentials and submit the form. A response with an `ok` field is expected.
+
+##### request data
+
+```js
+{
+  email: <email>
+}
+```
+
+##### response data
+
+```js
+{
+  ok: true | false;
+}
+```
+
+#### fxaccounts:login
+
+Sent when a user successfully authenticates with Firefox Accounts and sync can begin. This includes after a user signs up but before they verify their email address, after an account unlock, and after a password reset. No response is expected.
+
+##### data
+
+See [Login Data](#logindata).
+
+#### fxaccounts:oauth_login
+
+Sent when a user successfully authenticates via a WebChannel OAuth flow.
+
+##### data
+
+```
+{
+  "code": "02f3cfea84ac4c143662b38d6c7f0c82c6f91eb041befc7cecda446b1b4887c1",
+  "state": "vHao1p6OizzwReCkQMSpZA",
+  "redirect": "urn:ietf:wg:oauth:2.0:oob:oauth-redirect-webchannel",
+  "action": "signin",
+  "declinedSyncEngines": ["history"],
+  "offeredSyncEngines": ["history"]
+}
+```
+
+##### support
+
+Supported by `oauth_webchannel_v1` context.
+
+#### fxaccounts:verified
+
+The user has successfully verified their email address.
+
+##### data
+
+See [Login Data](#logindata).
+
+##### support
+
+#### fxaccounts:delete_account
+
+Sent after the user successfully deletes their account. No response is expected.
+Expected by Fennec.
```

Comment:
Ugh, I'd forgotten about this inconsistency between Fennec and Desktop. Since Fennec is going away Real Soon Now, is it worth filing a bug to remove this from FxA when it does?

Reply:
Once Fennec is replaced with Fenix, we should be able to remove the entire broker and all supporting functional tests! See https://github.com/mozilla/fxa/issues/3555